pubsub/natsstreaming/nats-streaming.go

```diff
@@ -0,0 +1,239 @@
+	} else {
+		return m, errors.New("nats-streaming error: missing nats URL")
+	}
+	if val, ok := meta.Properties[natsStreamingClusterID]; ok && val != "" {
+		m.natsStreamingClusterID = val
+	} else {
+		return m, errors.New("nats-streaming error: missing nats streaming cluster ID")
+	}
+
+	if val, ok := meta.Properties[subscriptionType]; ok {
+		if val == subscriptionTypeTopic || val == subscriptionTypeQueueGroup {
+			m.subscriptionType = val
+		} else {
+			return m, errors.New("nats-streaming error: valid value for subscriptionType is topic or queue")
+		}
+	} /*else {
+		return m, errors.New("nats-streaming error: missing subscriptionType")
+	}*/
+
+	if val, ok := meta.Properties[consumerID]; ok && val != "" {
+		m.natsQueueGroupName = val
+	} else {
+		return m, errors.New("nats-streaming error: missing queue group name")
+	}
+
+	if val, ok := meta.Properties[durableSubscriptionName]; ok && val != "" {
+		m.durableSubscriptionName = val
+	}
+
+	//subscription options - only one can be used
+	if val, ok := meta.Properties[startAtSequence]; ok && val != "" {
+		//nats streaming accepts a uint64 as sequence
+		seq, err := strconv.ParseUint(meta.Properties[startAtSequence], 10, 64)
+		if err != nil {
+			return m, fmt.Errorf("nats-streaming error %s ", err)
+		}
+		if seq < 1 {
+			return m, errors.New("nats-streaming error: startAtSequence should be equal to or more than 1")
+		}
+		m.startAtSequence = seq
+	} else if val, ok := meta.Properties[startWithLastReceived]; ok {
+		//only valid value is true
+		if val == startWithLastReceivedTrue {
+			m.startWithLastReceived = val
+		} else {
+			return m, errors.New("nats-streaming error: valid value for startWithLastReceived is true")
+		}
+	} else if val, ok := meta.Properties[deliverAll]; ok {
+		//only valid value is true
+		if val == deliverAllTrue {
+			m.deliverAll = val
+		} else {
+			return m, errors.New("nats-streaming error: valid value for deliverAll is true")
+		}
+	} else if val, ok := meta.Properties[deliverNew]; ok {
+		//only valid value is true
+		if val == deliverNewTrue {
+			m.deliverNew = val
+		} else {
+			return m, errors.New("nats-streaming error: valid value for deliverNew is true")
+		}
+	} else if val, ok := meta.Properties[startAtTimeDelta]; ok && val != "" {
+		dur, err := time.ParseDuration(meta.Properties[startAtTimeDelta])
+		if err != nil {
+			return m, fmt.Errorf("nats-streaming error %s ", err)
+		}
+		m.startAtTimeDelta = dur
+	} else if val, ok := meta.Properties[startAtTime]; ok && val != "" {
+		m.startAtTime = val
+		if val, ok := meta.Properties[startAtTimeFormat]; ok && val != "" {
+			m.startAtTimeFormat = val
+		} else {
+			return m, errors.New("nats-streaming error: missing value for startAtTimeFormat")
+		}
+	}
+
+	return m, nil
+}
+
+func (n *natsStreamingPubSub) Init(metadata pubsub.Metadata) error {
+	m, err := parseNATSStreamingMetadata(metadata)
+	if err != nil {
+		return err
+	}
+	n.metadata = m
+	clientID := genRandomString(20)
```

Comment:
Shouldn't this be consumerID? what is clientID used for?

Reply:
client ID is not the same as consumerID. Every client in NATS streaming needs to have a unique ID https://docs.nats.io/nats-streaming-concepts/client-connections

consumerID is used in the context of a queue group only